File: 1295_find_numbers_with_even_number_of_digits.py
class Solution(object):
    def findNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = 0
        for x in nums:
            d = 0
            while x > 0:
                x //= 10
                d += 1
            n += (d % 2 == 0)
        return n

vectors = [
        [12], 1,
        [12,345,2,6,7896], 2,
        [555,901,482,1771], 1
        ]

for i in range(0, len(vectors), 2):
    nums = vectors[i]
    expected = vectors[i + 1]
    print(f'{nums} {expected}')
    returned = Solution().findNumbers(nums)
    assert expected == returned, f'for {nums} there are {expected} numbers with even numbers, returned {returned}!'

File: C/1295_find_numbers_with_even_number_of_digits.c
int findNumbers(int* nums, int numsSize) {
    int n, d, x;
    if (!num || !numsSize)
        return 0;
    n = 0;
    for (int i = 0; i < numsSize; i++) {
        d = 0;
	x = nums[i];
	while (x > 0) {
            x /= 10;
	    d++;
	}
	n += (d % 2 == 0);
    }
    return n;
}
